rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py

```diff
@@ -67,13 +73,28 @@ class State:
     def __init__(self, state: RobotState = None, destination: Location = None):
         self.state = state
         self.destination = destination
+        self.svy_transformer = Transformer.from_crs('EPSG:4326', 'EPSG:3414')
+        self.gps_pos = [0, 0]
+
+    def gps_to_xy(self, gps_json: dict):
+        svy21_xy = \
+            self.svy_transformer.transform(gps_json['lat'], gps_json['lon'])
+        self.gps_pos[0] = svy21_xy[1]
+        self.gps_pos[1] = svy21_xy[0]
 
 
 class FleetManager(Node):
     def __init__(self, config, nav_path):
         self.config = config
         self.fleet_name = self.config["rmf_fleet"]["name"]
 
+        self.gps = False
+        self.offset = [0, 0]
+        if 'reference_coordinates' in self.config and \
+                'offset' in self.config['reference_coordinates']:
+            self.gps = True
+            self.offset = self.config['reference_coordinates']['offset']
```

Comment:
We should check that len(self.config['reference_coordinates']['offset']) > 1 before assigning to `self.offset`. Else when we reference self.offset[0] or self.offset[1] below, it could throw an out of range error. If the len < 1, throw an error and exit.

Reply:
:+1: